x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/process/autodetect/output/AutodetectResultProcessorTests.java

```diff
@@ -367,6 +380,23 @@ public void testProcessResult_modelSnapshot() {
         verify(persister).bulkPersisterBuilder(eq(JOB_ID), any());
         verify(persister).persistModelSnapshot(eq(modelSnapshot), eq(WriteRequest.RefreshPolicy.IMMEDIATE), any());
 
+        ArgumentCaptor<Annotation> annotationCaptor = ArgumentCaptor.forClass(Annotation.class);
+        verify(annotationPersister).persistAnnotation(
+            eq(ModelSnapshot.annotationDocumentId(modelSnapshot)), annotationCaptor.capture(), any());
+        Annotation annotation = annotationCaptor.getValue();
+        Annotation expectedAnnotation =
+            new Annotation(
+                "Job model snapshot with id [a_snapshot_id] stored",
+                Date.from(CURRENT_TIME),
+                XPackUser.NAME,
+                modelSnapshot.getTimestamp(),
+                modelSnapshot.getTimestamp(),
```

Comment:
I wonder if these should have been `modelSnapshot.getLatestResultTimeStamp()`?

See https://github.com/elastic/elasticsearch/issues/56076#issuecomment-622931389 for more details on this.

Reply:
I've just raised https://github.com/elastic/elasticsearch/pull/56093 to address that.